src/tests/test_lexical_cast.cpp

```diff
@@ -245,7 +245,11 @@ BOOST_AUTO_TEST_CASE(test_lexical_cast_result)
 	BOOST_CHECK_EQUAL(lexical_cast<int>("-1"), -1);
 	BOOST_CHECK_EQUAL(lexical_cast<unsigned>("1"), 1);
 	BOOST_CHECK_EQUAL(lexical_cast<double>("1.2"), 1.2);
-	BOOST_CHECK_THROW(lexical_cast<double>("0x11"), bad_lexical_cast);
+
+	// The unit [effect] code uses this a lot
+	BOOST_CHECK_EQUAL(lexical_cast_default<int>("80%"), 80);
+
+	BOOST_CHECK_EQUAL(lexical_cast<double>("0x11"), 0);
```

Comment:
This feels wrong, to be honest…

Reply:
It matches the behavior of "parse anything until an unknown char" of the 
'80%'  test above, and the attribute_value also explicitly tests for this being 0 in .to_int() .